handbook/engineering/code-intelligence/goals.md

```diff
@@ -25,46 +27,110 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
+    
+1. Increase C++ precise code intel monthly operations.
+    - Deliver to [*N<sub>1</sub>*][N1].
+    - Deliver to [*N<sub>2</sub>*][N2].
+    - Deliver to [*N<sub>3</sub>*][N3].
+    
+    **Outcome:** 3 enterprise customers with C++ precise code intel configured in their repositories.
 
 1. Sourcegraph Cloud has up-to-date precise code intelligence for C++ repositories that exceed 1mil SLOC.
-    - Reduce lsif-clang memory usage.
+    - Create benchmarks of lsif-clang memory and CPU usage for various sizes of repositories. 
     - Auto-index 5 repositories on Sourcegraph Cloud that exceed 1M SLOC.
     
     **Outcome:** 5 popular open source C++ repositories that exceed 1M SLOC have up-to-date precise code intelligence on Sourcegraph Cloud.
 
-1. Increase precise code intel monthly operations.
-    - Deliver to [*N<sub>1</sub>*][N1]
-    - Deliver to [*N<sub>2</sub>*][N2]
-    - Deliver to [*N<sub>3</sub>*][N3]
+1. Implement standard C++ code navigation features.
+    - Jump to declaration.
+    - Jump to implementation.
+    - Switch between source and header file.
+   
+
+### Java precise code intelligence support
+
+**Milestones:**
+
+1. Sourcegraph Cloud has up-to-date precise code intelligence for 5 Gradle repositories.
+    - Manual configuration in a forked repository.
+    - Gradle plugin implementation.
+    - Publish Gradle plugin.
+    - Update and improve documentation to ensure a good user experience.
+    
+    **Outcome:** 5 popular open source Gradle repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
+
+1. Sourcegraph Cloud has up-to-date precise code intelligence for 5 Maven repositories.
+    - Manual configuration in a forked repository.
+    - Maven plugin implementation.
+    - Publish Maven plugin.
+    - Update and improve documentation to ensure a good user experience.
     
-    **Outcome:** The C++ indexer is adopted by at least 3 customers.
-    **Outcome:** Precise code intel reaches [*N<sub>0</sub>*][N0] precise code intel operations a month. 
+    **Outcome:** 5 popular open source Maven repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
+ 
+1. Sourcegraph Cloud has precise code intelligence for Java repositories that use Bazel as a build system.
+    - Manual configuration in a forked repository.
+    - Bazel script implementation.
+    - Update and improve documentation to ensure a good user experience.
+    
+    **Outcome:**  5 popular open source Java repositories using Bazel have up-to-date precise code intelligence on Sourcegraph Cloud.
+
+1. Increase Java precise code intel monthly operations.
+    - Deliver to [*N<sub>4</sub>*][N4].
+    - Deliver to [*N<sub>5</sub>*][N5].
+    - Deliver to [*N<sub>6</sub>*][N6].
+    
+    **Outcome:** 3 enterprise customers with Java precise code intel configured in their repositories.
+    
+1. Lsif-java supports third party dependency navigation.
+
+1. Lsif-java supports Java 11-16.
```

Comment:
Im not sure which should come first here, or if this isnt a strict ordering. I feel supporting 11+ would be a quicker win than third party navigation, which afaik relies on RFC 307 which is a big task :smile: if we get wider JDK support we might be able to get local intel out to a few more customers/repos and then iterate upon that by working on library navigation. Im not too concerned about this but just thought it put my 2c out there @olafurpg @macraig

Reply:
@Strum355 it's a good point, especially considering our conversation about support for 11+ potentially being a quick win. Just to add some context, these are tentatively ordered and we can change them (now or further in the future) if it makes sense. Would you and @olafurpg rather leave it this way or switch it?